DEVELOPING.md

```diff
@@ -0,0 +1,126 @@
+# Logstash Development Workflow
+
+This explains how to develop in the logstash core code-base. We will use to following fictional directories:
+
+- `/logstash/` is the dir containing your local fork clone of logstash
+- `/plugins/...` is the dir containing your local fork clones of logstash plugins or any other local private plugins
+
+## Setup
+
+Logstash uses [JRuby](http://jruby.org/) which gets embedded in the `vendor/jruby/` directory. It is **highly recommended** but not mandatory that you also use JRuby as your local Ruby interpreter and for this you should consider using a Ruby version manager such as [RVM](https://rvm.io/) or [rbenv](https://github.com/sstephenson/rbenv). It is possible to run the rake tasks and the `bin/` commands without having JRuby locally installed in which case the embedded JRuby will be used automatically.
+
+You will also need to have a JVM installed, we recommend using Java 1.8.
+
+`Rake` and `Bundler` should also be installed in your environment.
+
+Note that if you have a local JRuby installed you can force logstash to use your local JRuby instead of the embedded JRuby with the `USE_RUBY=1` environment variable.
+
+
+## Bootstrapping
+
+The logstash environment need to be bootstrapped to install base dependencies in the `vendor/` directory. Use the following command to bootstrap:
+
+```sh
+$ rake bootstrap
+```
+
+After bootstrap, logstash can be launched using `bin/logstash` but note that at this point, no plugin is installed.
+
+### Installing Core Plugins
+
+The logstash core plugins are required to run the core tests/specs. These core plugins and their development dependencies can be installed using:
+
+```sh
+$ rake test:install-core
```

Comment:
yes and no, there is actually a task for this in the plugin namespace, but this one is also doing bootstrap and other things, see https://github.com/elastic/logstash/blob/master/rakelib/test.rake#L75. I think this an important separation of concerns, what we might improve probably is naming, but not sure how...